src/sugarblas.jl

```diff
@@ -0,0 +1,143 @@
+#####################
+# Blas Completition #
+#####################
+import Base.LinAlg.BLAS: blascopy!, scal!, scal
+
+# Add robustness to blascopy!
+function blascopy!(n::Number, DX::AbstractArray, incx::Integer, DY::AbstractArray, incy::Integer)
+    x = 1
+    y = 1
+    nx = min(n, length(DX))
+    ny = min(n, length(DY))
+    while x <= nx && y <= ny
+        DY[y] = DX[x]
+        x += incx
+        y += incy
+    end
+    DY
+end
+
+# Add robustness to scal!
+function scal!(n::Integer, DA::Number, DX::AbstractArray, incx::Integer)
```

Comment:
Since you are checking array lengths and the loops are simple don't you think `@inbounds` and `@simd` could be good for performance?